Remove unused setup.py keywords "test_suite"

jdufresne opened the pull request:
The tests are run through either tox or Travis CI. In both cases, the
setup.py "test_suite" entrypoint is not used.

Further, the use of "test_suite" is discouraged as it installs using
eggs rather than pip.

jdufresne commented:
Why keep it? it is unused.

Locally, to test one can use `tox` or `tox -e py38`. On Travis, the `.travis.yml` file does its own thing. So I don't see the point of adding a different test entry point.

Are you using this now? It doesn't run the `simple_unit_tests.py` script.

klahnakoski commented:
@ptmcg My understanding is we are keeping this line, but I am also glad to be wrong.  I defer to your authority

ptmcg commented:
I had a request to support "python setup.py test" (though I can't track it down at the moment), and this was the change that was needed to support that.  This looks like a pretty standard way for people to run  tests on packages they've pulled from GitHub, in addition to other available test mechanisms like tox, et al.  I'm inclined to leave it, and I think Kyle's changes pull in some other tests too(?).

jdufresne commented:
The setuptools maintainers recommended against using the `python setup.py test` and the `test_suite` keyword. They consider them both deprecated. They recommend instead to use tox, which is what I'm suggesting with this change -- that we follow their recommendation.

Here are their statements on the matter:

https://github.com/pypa/setuptools/issues/931
https://github.com/pypa/setuptools/issues/1684

ptmcg commented:
Would be good to add a note to CHANGES file that 'python setup.py test' is no longer supported, with the pypa links or quotes included.

jdufresne commented:
Done. Added an entry to CHANGES. I also added `tox.ini` to the sdist through `MANIFEST.in` so distro packagers can continue to run tests with a single command.

ptmcg commented:
Please resolve the conflicts on this PR, then I can merge it.

jdufresne commented:
Done
